src/Exercise/AbstractExercise.php

```diff
@@ -35,15 +36,19 @@ abstract public function getName(): string;
      */
     public function getSolution(): SolutionInterface
     {
-        return SingleFileSolution::fromFile(
-            (string) realpath(
-                sprintf(
-                    '%s/../../exercises/%s/solution/solution.php',
-                    dirname((string) (new ReflectionClass(static::class))->getFileName()),
-                    self::normaliseName($this->getName())
+        if (null === $this->solution) {
```

Comment:
so this was required and probs also for dir solution... the problem being that a patched solution in temp was being overwritten due to the solution being copied on construct and we're constructing fresh on each call to getSolution.

Probably a better fix here, maybe moving the copy to temp dir to somewhere that isn't on solution construct.

Reply:
@AydinHassan I need your input on this one bro, can you think of a cleaner approach that for this?

Reply:
Hmm not really off the top of my head. I think it's fine as you have it.

Reply:
I guess the only problem I foresee here is the implementation of a solution inside the exercise instead of using the Abstract. e.g if you need a dir solution and you use the static factory 🤔